aws/config.go

```diff
@@ -208,8 +208,22 @@ type Config struct {
 	//     svc := s3.New(sess, &aws.Config{
 	//         UseDualStack: aws.Bool(true),
 	//     })
+	//
+	// Deprecated: This option will continue to function for S3 and S3 Control for backwards compatibility.
+	// DualStackEndpoint should be used to enable usage of a service's dual-stack endpoint for all service clients
+	// moving forward. For S3 and S3 Control, when DualStackEndpoint is set to a non-zero value it takes higher
+	// precedence then this option.
 	UseDualStack *bool
```

Comment:
No this would be backwards in compatible unfortunately as users may have been setting this globally previously. Just it would have no effect on clients other then s3 or s3 control. This would cause the other clients to start failing unexpectedly.